test/python/conftest.py

```diff
@@ -1,3 +1,111 @@
+import cantera
+from os import environ
+from pathlib import Path
 import pytest
+import tempfile
+
+try:
+    from ruamel import yaml
+except ImportError:
+    import ruamel_yaml as yaml
 
 pytest.register_assert_rewrite("pint.testing")
+
+
+TEST_DATA_PATH = Path(__file__).parents[1] / "data"
+CANTERA_DATA_PATH = Path(cantera.__file__).parent / "data"
+
+
+def pytest_configure(config):
+    config.addinivalue_line("markers", "slow_test: mark test as slow")
+
+def pytest_collection_modifyitems(config, items):
+    if environ.get("CT_SKIP_SLOW", "0") == "1":
+        skip_slow = pytest.mark.skip(reason="slow test")
+        for item in items:
+            if "slow_test" in item.keywords:
+                item.add_marker(skip_slow)
+
+@pytest.fixture
+def allow_deprecated():
+    cantera.suppress_deprecation_warnings()
+    yield
+    cantera.make_deprecation_warnings_fatal()
+
+
+@pytest.fixture
+def has_temperature_derivative_warnings():
+    with pytest.warns(UserWarning, match="ddTScaledFromStruct"):
+        # test warning raised for BlowersMasel and TwoTempPlasma derivatives
+        yield
+
+@pytest.fixture(scope="session")
+def test_data_path():
+    return TEST_DATA_PATH
+
+@pytest.fixture(scope="session")
+def cantera_data_path():
+    return CANTERA_DATA_PATH
+
+@pytest.fixture(scope="session", autouse=True)
+def cantera_setup():
+    """
+    Fixture to set up Cantera environment for the entire test session.
+    """
+    # Add data directories
+    cantera.add_directory(TEST_DATA_PATH)
+    cantera.add_directory(CANTERA_DATA_PATH)
+    cantera.print_stack_trace_on_segfault()
+    cantera.CanteraError.set_stack_trace_depth(20)
+    cantera.make_deprecation_warnings_fatal()
+
+    # Yield control to tests
+    yield
```

Comment:
I may be wrong about what's idiomatic for pytest, but it seems like the simplest thing to do would be to execute these functions directly at the module level, as done in the old `utilities.py`. The use case for the session-level fixture seems like it would be if it created a singleton object of some sort to actually be used in other some other tests, but here you just want the side-effects that affect the global module state.

This should fix the CI failures. I think the reason you don't see them locally is if you're running the tests through SCons, it's already adding `test/data` to the cantera data path. The CI runs pytest directly, so has to rely on this for the path update.